docs/2.10.0/docs/canton/usermanual/security.rst

```diff
@@ -320,6 +320,29 @@ A fingerprint of a key can be retrieved from the list of public keys:
    :end-before: user-manual-entry-end: ListPublicKeys
    :dedent:
 
+.. _deleting-canton-keys:
+
+Deleting Canton Node Keys
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Deletion of keys, in particular, private keys is not done automatically in Canton. When a key is rotated, the previous
+key remains stored, but is disabled and no longer used. To permanently delete a key, use the following Canton console
+command:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/scala/com/digitalasset/canton/integration/tests/security/topology/KeyManagementIntegrationTest.scala
+   :language: scala
+   :start-after: user-manual-entry-begin: DeletePrivateKey
+   :end-before: user-manual-entry-end: DeletePrivateKey
+   :dedent:
+
+.. warning::
+  Exercise caution when deleting a private key. You must ensure that the key is no longer in use and is not needed
+  for decrypting or verifying signatures of previous messages. Therefore, keys must only be deleted after you have
+  pruned all data that directly or indirectly uses them. This is especially crucial for sequencers in open networks.
```

Comment:
There is no guidance on how an operator can figure out whether all data has been pruned that directly or indirectly uses them. A conservative rule of thumb would be to say that you should only delete the key when the node was pruned for a timestamp that is later than the point in time when the key was deactivated.